### PR TITLE
implements concordance_index for regression evaluation

### PR DESCRIPTION
**Pull request by hsjang001205**
Similar to AUC in regression. Evaluate values for distributed differences, not accuracy.
As an example, if the IC50 value for drug A and B is each 30 and 40, the predicted value for A should be lower than B.
It is a value of whether there is a significant distributional difference with the ranking information.
(often used in clinical data.)

(Lifelines provides more extended functions, so it would be a good idea to consider that.)

### CODE REVIEW

File: deepchem/metrics/score_function.py [hunk 1]
```diff
@@ -162,3 +162,54 @@ def bedroc_score(y_true: np.ndarray, y_pred: np.ndarray, alpha: float = 20.0):
...
+  Parameters
+  ----------
+  y_true: np.ndarray
+    continous value
+  y_pred: np.ndarray
+    Predicted value
+
+  Returns
+  -------
+  float between [0,1]
```
Comment (nd-02110114) [inline deepchem/metrics/score_function.py:182]: The docstring is not appropriate. The following is the correct style.
```
Returns
-------
int
    Description of anonymous integer return value.
```

**Review by nd-02110114 — COMMENTED** [review]
Thank you for contribution!
I just commented the docstring for the return value.

File: deepchem/metrics/score_function.py (hunk 1 above)
Comment (hsjang001205) [reply deepchem/metrics/score_function.py]: Ah, I checked.

File: deepchem/metrics/score_function.py [hunk 2]
```diff
@@ -162,3 +162,55 @@ def bedroc_score(y_true: np.ndarray, y_pred: np.ndarray, alpha: float = 20.0):
   scores = sorted(scores, key=lambda pair: pair[1], reverse=True)
 
   return CalcBEDROC(scores, 0, alpha)
+
+
+def concordance_index(y_true: np.ndarray, y_pred: np.ndarray) -> float:
```
Comment (rbharath) [inline deepchem/metrics/score_function.py:167]: Can you add a autodoc for this new metric to `docs/metrics.rst` along with the other metrics there? That will ensure `concordance_index` shows up in our documentation

**Review by rbharath — COMMENTED** [review]
@hsjang001205 Looks good! Almost ready to merge in. Have one minor request about adding this to the docs and we should be good to merge in

**Review by rbharath — APPROVED** [review]
LGTM! Feel free to merge when ready :)

**Comment by nd-02110114** [comment]
I merge in. This is because this PR touch docs, so the conflict will happen if my #2257 merge ahead.

**Pull request merged**